tests/e2e/device-plugin_test.go

```diff
@@ -108,9 +127,19 @@ var _ = NVDescribe("GPU Device Plugin", func() {
 
 		// Cleanup before next test run
 		AfterEach(func(ctx context.Context) {
-			// Gather logs
+			// Run diagnostic collector if test failed
 			if CurrentSpecReport().Failed() {
-				_ = common.MustGather("k8s-device-plugin", f.Namespace.Name, filepath.Join(*LogArtifactDir, f.UniqueName))
+				var err error
+				diagnostic, err = diagnostics.New(
+					diagnostics.WithNamespace(f.Namespace.Name),
+					diagnostics.WithArtifactDir(*LogArtifactDir),
+					diagnostics.WithKubernetesClient(f.ClientSet),
+					diagnostics.WithObjects(collectLogsFrom...),
+				)
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
I think we also just want to log this error as we do for the call to `Collect()` below.